File: Challenges/Problem_9/Special_Pythagorean_triplet.js
// KuntaNgcobo
// Special Pythagorean triplet

const a = (x, y) => { return y*y - x*x };
const b = (x, y) => { return 2*x*y };
const c = (x, y) => { return y*y + x*x };

const specialPythagoreanTriplet = n => {
 for(var i = 1; i < n+1; i++){
   for(var j = 1; j < i; j++){
       let diff = n - a(j,i) - b(j,i) - c(j,i);
       if(diff === 0){
           return a(j,i) * b(j,i) * c(j,i)
       }
   }
 }
 return -1;
}

​

File: Challenges/Problem_9/Special_Pythagorean_triplet.py
# KuntaNgcobo
# Special Pythagorean triplet

a = lambda x, y : y*y - x*x
b = lambda x, y : 2*x*y
c = lambda x, y : y*y + x*x

def specialPythagoreanTriplet(n):
	for i in range(1, n+1):
	 	for j in range(1, i):
	 		diff = n - a(j,i) - b(j,i) - c(j,i)
	 		if(diff == 0):
	 			return a(j,i) * b(j,i) * c(j,i)
	return -1
